modules/serverless-deploy-func-kn.adoc

```diff
@@ -0,0 +1,32 @@
+// Module included in the following assemblies:
+//
+// serverless/serverless-functions-quickstart.adoc
+// add user guide?
+
+[id="serverless-deploy-func-kn_{context}"]
+= Deploying functions
+
+You can deploy a function to your cluster as a Knative service, by using the `kn func deploy` command.
+
+If the targeted function is already deployed, it is updated with a new container image that is pushed to a container image registry, and the Knative service is updated.
+
+[NOTE]
+====
+If you are using `quay.io` as the image registry, you must ensure that the repository is not private. The first time an image is deployed these settings may need to be updated.
+====
```

Comment:
Well, I was able to deploy functions with private quay.io repositories, by adding the login credentials to OpenShift using image pull secrets (same procedure explained here):

https://docs.openshift.com/container-platform/4.7/openshift_images/managing_images/using-image-pull-secrets.html#images-allow-pods-to-reference-images-from-secure-registries_using-image-pull-secrets

As suggestion, probably, this option could be mentioned as well.

Reply:
@lance wdyt?

Reply:
Yes - absolutely @abrennan89! That's an excellent point @jrangelramos.

Reply:
Mentioned this in the assembly as part of prereqs rather than here because we can't include xref links in modules yet, lmk what you think 😄 thanks @jrangelramos for the great suggestion